WarpBuf/src/WarpBufChugin.cpp

```diff
@@ -1,448 +1,146 @@
-
-//-----------------------------------------------------------------------------
-// name: class WarpBufChugin
-// desc: WarpBufChugin for time-stretching and pitch-stretching (via Rubberband library)
-//-----------------------------------------------------------------------------
-class WarpBufChugin
+WarpBufChugin::WarpBufChugin(t_CKFLOAT srate)
 {
-public:
-    // constructor
-    WarpBufChugin(t_CKFLOAT srate)
-    {
-        // sample rate
-        m_srate = srate;
-        memset(&sfinfo, 0, sizeof(SF_INFO));
-
-        using namespace RubberBand;
-
-        RubberBandStretcher::Options options = 0;
-        options |= RubberBandStretcher::OptionProcessRealTime;
-        options |= RubberBandStretcher::OptionStretchPrecise;
-        //options |= RubberBandStretcher::OptionPhaseIndependent;
-        //options |= RubberBandStretcher::OptionWindowLong;
-        //options |= RubberBandStretcher::OptionWindowShort;
-        //options |= RubberBandStretcher::OptionSmoothingOn;
-        //options |= RubberBandStretcher::OptionFormantPreserved;
-        options |= RubberBandStretcher::OptionPitchHighQuality;
-        //options |= RubberBandStretcher::OptionChannelsTogether;
-
-        // Pick one of these:
-        options |= RubberBandStretcher::OptionThreadingAuto;
-        //options |= RubberBandStretcher::OptionThreadingNever;
-        //options |= RubberBandStretcher::OptionThreadingAlways;
-
-        // Pick one of these:
-        options |= RubberBandStretcher::OptionTransientsSmooth;
-        //options |= RubberBandStretcher::OptionTransientsMixed;
-        //options |= RubberBandStretcher::OptionTransientsCrisp;
-
-        // Pick one of these:
-        options |= RubberBandStretcher::OptionDetectorCompound;
-        //options |= RubberBandStretcher::OptionDetectorPercussive;
-        //options |= RubberBandStretcher::OptionDetectorSoft;
-
-        m_rbstretcher = std::make_unique<RubberBand::RubberBandStretcher>(
-            m_srate,
-            2,
-            options,
-            1.,
-            1.);
-
-        m_nonInterleavedBuffer = new float * [channels];
-        for (int i = 0; i < channels; i++) {
-            m_nonInterleavedBuffer[i] = new float[ibs];
-        }
-        m_interleavedBuffer = new float[channels * ibs];
+    // sample rate
+    m_srate = srate;
+    memset(&sfinfo, 0, sizeof(SF_INFO));
 
-    }
+    this->resetStretcher();
+}
 
-    ~WarpBufChugin()
-    {
-        clearBufs();
-        delete[] m_interleavedBuffer;
+void
+WarpBufChugin::resetStretcher() {
 
-        if (m_nonInterleavedBuffer != NULL)
-        {
-            for (int i = 0; i < channels; i++) {
-                SAFE_DELETE_ARRAY(m_nonInterleavedBuffer[i]);
-            }
-        }
-        SAFE_DELETE_ARRAY(m_nonInterleavedBuffer);
+    using namespace RubberBand;
 
-        m_rbstretcher.release();
-    }
+    RubberBandStretcher::Options options = 0;
 
+    //options |= RubberBandStretcher::OptionProcessOffline;
```

Comment:
could you clean up these comments?

Reply:
Thanks. I've updated. It might be nice to have some enum system for this.